packages/overlay/src/vaadin-overlay-mixin.js

```diff
@@ -175,12 +175,15 @@ export const OverlayMixin = (superClass) =>
      * @param {Event=} sourceEvent
      */
     close(sourceEvent) {
+      // Dispatch the event on the overlay. Not using bubbles or composed, as that could have side effects when nesting
+      // overlays
       const evt = new CustomEvent('vaadin-overlay-close', {
-        bubbles: true,
         cancelable: true,
         detail: { sourceEvent },
       });
       this.dispatchEvent(evt);
+      // To allow listening for the event globally, also dispatch it on the document body
+      document.body.dispatchEvent(evt);
```

Comment:
Maybe the events on the body should bubble, so that it reaches `document` as well?

Reply:
Yes, bubbling to `document` sounds reasonable. 

BTW do we really need to remove `bubbles`? Without `composed: true` the event should only bubble up to shadow root, and will not reach the host anyway.

Reply:
It's probably OK to keep it, makes propagation to `document` easier. I was worried that it might cause side-effects if the DOM changes again, but that seems unlikely.

So I restored bubbles, updated tests to listen on `document` and changed the new test names to reflect that the events should not propagate through shadow roots.